### PR TITLE
fix(createtable): pass table field name in attr

### LINKED ISSUE #11285: queryInterface.createTable() does not work with ARRAY(ENUM) with defaultValue
## What are you doing?

In postgres:

```js
await sequelize.queryInterface.createTable('test', {
    test: {
        type: Sequelize.ARRAY(Sequelize.ENUM(['foo', 'bar', 'baz', 'qux'])),
        allowNull: false,
        defaultValue: ['foo']
    }
});
```

**To Reproduce:** Just run the snippet above, no models or other setup needed
2. See error: `TypeError: Cannot read property 'getTableName' of undefined`

## What do you expect to happen?
It should work

## What is actually happening?

`TypeError: Cannot read property 'getTableName' of undefined`

Thrown at line:

https://github.com/sequelize/sequelize/blob/748f69080fdf86b8968e14c02e330b2a1cf41ebc/lib/dialects/postgres/data-types.js#L481

## Environment
Dialect:
- [x] postgres

Tested with latest release:
- [x] Yes, specify that version: 5.12.3

## Extra info

If the `defaultValue` is removed, it works:

```js
await sequelize.queryInterface.createTable('test', {
    test: {
        type: Sequelize.ARRAY(Sequelize.ENUM(['foo', 'bar', 'baz', 'qux'])),
        allowNull: false
    }
});
```

### PR DESCRIPTION
**Pull request by krazi3**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change
Closes #11285

Set `tableName` and `fieldName` on attribute with type ARRAY(ENUM)

### CODE REVIEW

File: lib/dialects/postgres/query-generator.js [hunk 1]
```diff
@@ -476,6 +476,12 @@ class PostgresQueryGenerator extends AbstractQueryGenerator {
 
         if (attribute.type instanceof DataTypes.ARRAY) {
           type += '[]';
+
+          // Add table and field name if not exists on attributes with ARRAY(ENUM) type
+          if (!attribute.fieldName) {
+            attribute.fieldName = options.key;
+            attribute.tableName = options.table.tableName || options.table;
```
Comment (sushantdhiman) [inline lib/dialects/postgres/query-generator.js]: `attribute.tableName` is not used anywhere else, I think.

File: lib/dialects/postgres/data-types.js [hunk 2]
```diff
@@ -478,7 +478,7 @@ module.exports = BaseTypes => {
 
       if (this.type instanceof BaseTypes.ENUM) {
         castKey = `${Utils.addTicks(
-          Utils.generateEnumName(options.field.Model.getTableName(), options.field.fieldName),
+          Utils.generateEnumName(options.field.tableName || options.field.Model.getTableName(), options.field.fieldName),
```
Comment (sushantdhiman) [inline lib/dialects/postgres/data-types.js]: Isn't `options.table.tableName` available here? It seems odd for attribute to keep `tableName`

Comment (krazi3) [reply lib/dialects/postgres/data-types.js]: I don't think i get the `table` object in options ... i will check once again

File: lib/dialects/postgres/query-generator.js (hunk 1 above)
Comment (krazi3) [reply lib/dialects/postgres/query-generator.js]: Yes this looked a bit weird to me as well ... but i had to pass table info at
https://github.com/sequelize/sequelize/pull/11517/files#diff-7e347c4738c97261969fbeb802a3c363R511

**Comment by WikiRik** [comment]
Closing this since it is still against `master`. Can be reopened if that changes.

**Pull request closed without merging**